pkg/imgpkg/bundle/bundle_images_lock.go

```diff
@@ -35,7 +35,12 @@ func (o *Bundle) AllImagesRefs(concurrency int, logger util.LoggerWithLevels) ([
 			if !found {
 				panic(fmt.Sprintf("Internal inconsistency: The Image '%s' cannot be found in the total list of images", ref.Image))
 			}
-
+			imgRef = imgRef.DeepCopy()
+			parsedImageRefName, err := regname.ParseReference(ref.Image)
```

Comment:
if we are changing this stuff here this would affect places that end up reading stuff (second copy or pull). i dont think added test accounts for that currently.

Reply:
> affect places that end up reading stuff

affect in a negative way? I fail to see how using a fully qualified name would cause issues? 

> second copy or pull

We have an e2e test that effectively does this scenario already and runs assertions on the oci location file too.